src/m3ninx/search/searcher/conjunction.go

```diff
@@ -72,13 +87,20 @@ func (s *conjunctionSearcher) Search(r index.Reader) (postings.List, error) {
 		}
 	}
 
+	lists = lists[:0]
```

Comment:
Should we add an early return if `pl` is empty after processing intersections?

Reply:
I think we can achieve this by simply moving those `if pl.IsEmpty()` checks to the top of `for` loops which makes sense in general (and let's not worry about avoiding the sort).

Reply:
https://github.com/m3db/m3/pull/3981/commits/cdffc644c692b40081c5e04dc881bc7c13f6f789